### PR TITLE
multiple fixes to make helm work

### LINKED ISSUE #5733: Can't disable traefik ingress controller without removing ingress object
Hi!

We have the next use case: we are using k8s with already installed nginx ingress controller. In fact it can be any ingress controller of our choice. As well - it may be already installed traefik. Right now I can't disable installation of traefik embedded into CVAT helm chart because in that case Ingress object won't be created. 

https://github.com/opencv/cvat/blob/27224f25967d5e13ade686c4dc7f1bd363415d74/helm-chart/Chart.yaml#L61
https://github.com/opencv/cvat/blob/27224f25967d5e13ade686c4dc7f1bd363415d74/helm-chart/values.yaml#L289

### The expected behaviour:

to have two options in helm chart:

1. disable of Ingress object itself
2. disable of embedded Traefik ingress controller in case when customer doesn't want it.

My proposal is to make another enabled option under traefik block in values.yaml:

```yaml
traefik:
  enabled: true # default value if you want to install Traefik by default.
  logs:
    general:
      level: INFO
  service:
    externalIPs:
      # - "192.168.49.2"
```

and to change traefik block in Chart.yaml:

```yaml
  - name: traefik
    version: 10.24.0
    repository: https://helm.traefik.io/traefik
    condition: traefik.enabled
``` 

The change is trivial, so I may create a PR for it.

### LINKED ISSUE #6043: [Helm] multiples fixes needed
Hi, I cannot create another PR right now, or update current one, but I decided at least to write down my chart concerns in case someone else could fix that:

1. Treafik is hardcoded to values right now and that is bad, because it is not standard package and people could use other software as ingress(nginx/istio/ambassador/etc)
2. Previously there was a way to specify psql db creds via secret, that was for some reason removed in new versions. That is important because some people(including me) using Terraform to create db in cloud and provirde k8s secret to pass creds to cvat
3. Right now I see, that there is 5-6 deployments of cvat deployed by default - are you sure, that I need all of them just to start cvat on my cluster? My guess that at least part of them should be opt-in and switched off by default. I see that compose with https uses only one!
4.  Right now there is only one PVC for all of the deployments, I think that will not work from the box for almost anybody, who tried to use this chart

### LINKED ISSUE #6096: Helm Chart Volume Configuration "ReadWriteOnce" Doesn't Work with Multiple Pods
### My actions before raising this issue
- [X] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [X] Searched [past issues](/issues)

When using the Helm Chart, the storage.yml ReadWriteOnce option under the backend folder structure doesn't work in my K8s environment. If you have deployments across multiple nodes, ReadWriteOnce will form a race condition causing only one backend pod to start successfully. ReadWriteOnce by definition only allows one node to mount - so how is this supposed to work? 

### Steps to Reproduce (for bugs)
1. Attempt to deploy the helm chart to Kubernetes with the default storage

### Expected Behaviour
1. The defaults for the helm chart should be able to launch CVAT in K8s with the minimal amount of modification. 
### Possible Solution
1. Changing storage.yml to ReadWriteMany 
2. Put all backend deployments into the same node (maybe there's a reason this isn't designed this way though)

### PR DESCRIPTION
**Pull request by Keramblock**
### Motivation and context

Right now helm chart is broken and not usable at least in my environment, I trying to fix it to make it work
content: 

1. Moved test-related values to another values.file to separate it from default config
2. fixed issue with multiple caches in same RWX volume, which prevents db migration to start
3. Removed hardcoded mandatory traefik ingress usage
4. Added confugurable default storage option to chart

### How has this been tested?

We test it on our AKS with RWX volume

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
  
closes #6043 
closes #6096 
closes #5733

### CODE REVIEW

**Comment by Keramblock** [comment]
@mdacoca if you could also help with the review and helm tests - that would be great =)

**Comment by azhavoro** [comment]
@Keramblock Hi, Thanks for the contributing! I will test the PR this week.

**Comment by Keramblock** [comment]
@azhavoro have you had a chance to take a look?

File: helm-chart/values.yaml [hunk 1]
```diff
@@ -91,6 +91,9 @@ cvat:
             name: http
     defaultStorage:
         enabled: true
+        storageClassName: default
+        accessModes:
+         - ReadWriteOnce
```
Comment (DreamingRaven) [inline helm-chart/values.yaml:96]: Correct me if I am wrong, I was testing the cvat helm chart out today, and I believe this should probably be ReadWriteMany as at-least the original chart before your changes would spin up multiple pods seeking to access the same volume by PVC name. I dont know how cvat acts when multiple pods can read and write at once, but without this set to ReadWriteMany only one of the pods like utils or server will get this volume and the rest will be waiting for it indefinitely since they cant bind to it any more.

Comment (Keramblock) [reply helm-chart/values.yaml:96]: Hi, I have limited knowledge of what is happening internally, so I left it as it is and I'm waiting for input from @azhavoro or @mdacoca to see if it's okay to change it by default.

In my opinion, the current chart may not work as it is, and we might need to change the default access mode. However, I'm almost certain that it would not help because the default storage class usually does not support this access mode either.

P.S. Thank you very much for reviewing! I appreciate it.

Comment (DreamingRaven) [reply helm-chart/values.yaml:96]: Ah I see, for me I back my kubernetes cluster with rook so I have a default storage class that supports RWX but yeah I get what you mean, there does not seem to be a good solution for everyone but I think RWO is broken for everyone.

Also on a related note I might not assume the users default storage class, In my case I use [rook](https://rook.io/) which gives me these:
```
NAME                        PROVISIONER                     RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
ceph-block                  rook-ceph.rbd.csi.ceph.com      Delete          Immediate           true                   26h
ceph-bucket                 rook-ceph.ceph.rook.io/bucket   Delete          Immediate           false                  26h
ceph-filesystem (default)   rook-ceph.cephfs.csi.ceph.com   Delete          Immediate           true                   26h
```
Im not 100% certain but I believe if in the PVC you dont specify the storage class that it will auto populate it with whatever the cluster has as a default no matter what it is called. This might be more universal if you put this behind and if condition so if the user sets a storageClassName it will use it, and if not it will leave it blank in the CRD for kubernetes to fill it in. Something like {{ if eq .Values.cvat.backend.defaultStorage.storageClassName "" }} maybe or {{ if empty .Values.cvat.backend.defaultStorage.storageClassName }} (this was only quick off the top of my head may need editing the condition to invert it).

Also I wanted to say its no problem, I have a vested interest in this too so I would love to see your PR be merged as it works a lot better than the current one which is a bit broken in a few places.

Comment (Keramblock) [reply helm-chart/values.yaml]: Thanks a lot, added)

Comment (DreamingRaven) [reply helm-chart/values.yaml]: That's beautiful thanks for the changes!!! This now has made ArgoCD happier as well since it understands the value is now auto-filled by kube. Thanks a lot for your additions, super appreciated!
cvat
I will keep testing the chart for you and let you know if there are any other minor changes but it all looks good currently.

File: helm-chart/values.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ cvat:
       additionalVolumeMounts: []
     worker:
       export:
-        replicas: 2
+        replicas: 1
```
Comment (azhavoro) [inline helm-chart/values.yaml:27]: Why did you change the default value? From my point of view, it is better to have a similar number of workers for k8s and docker-compose

File: helm-chart/values.yaml [hunk 3]
```diff
@@ -34,7 +34,7 @@ cvat:
         additionalVolumes: []
         additionalVolumeMounts: []
       import:
-        replicas: 2
+        replicas: 1
```
Comment (azhavoro) [inline helm-chart/values.yaml:37]: the same comment

File: helm-chart/values.yaml [hunk 4]
```diff
@@ -225,7 +228,7 @@ nuclio:
 #      password: somepass
 
 analytics:
-  enabled: true
+  enabled: false
```
Comment (azhavoro) [inline helm-chart/values.yaml:231]: Analytics is currently enabled by default, please revert the change

File: helm-chart/values.yaml [hunk 5]
```diff
@@ -319,62 +322,58 @@ grafana:
...
-            name: backend-service
-            port: 8080
-        - path : /
-          pathType: "Prefix"
-          service:
-            name: frontend-service
-            port: 80
+  enabled: false
+#  enabled: true
+#  annotations:
```
Comment (azhavoro) [inline helm-chart/values.yaml:327]: `ingress.enabled = true` does nothing, why rules should be commented by you opinion?

**Review by azhavoro — APPROVED** [review]

File: helm-chart/values.yaml (hunk 4 above)
Comment (Keramblock) [reply helm-chart/values.yaml:231]: will do

File: helm-chart/values.yaml (hunk 5 above)
Comment (Keramblock) [reply helm-chart/values.yaml:327]: Because there are some lists in that config and they are hard to override with user values files.
For example "paths" list  here will be very hard to override for me to use another Ingress provider like nginx, or ambassador, or gke lb

Comment (Keramblock) [reply helm-chart/values.yaml:327]: my config just for example: 
image

Comment (Keramblock) [reply helm-chart/values.yaml:327]: also ingress.enblaed is used here, for example: 
изображение

Comment (Keramblock) [reply helm-chart/values.yaml:327]: also ingress.enabled is used here: 
изображение

File: helm-chart/values.yaml (hunk 2 above)
Comment (eremeevfd) [reply helm-chart/values.yaml:27]: We can drop a comment why do we have 2 replicas by default maybe? Just for people to know what it is and why they should or shouldn't change it

File: helm-chart/values.yaml (hunk 5 above)
Comment (eremeevfd) [reply helm-chart/values.yaml:327]: @azhavoro @Keramblock Maybe we should add ImplementationSpecific to default values?

Comment (eremeevfd) [reply helm-chart/values.yaml:327]: @Keramblock Could you please add comment to commented values in ingress section, smth like "In case you need an ingress, don't forget to uncomment paths, or write them manually in your values"

Comment (Keramblock) [reply helm-chart/values.yaml:327]: @azhavoro also I moved traefik configuration to tests.values.yml just in case you will need it for your internal stuff, so that change will not interrupt you workflow.

Comment (eremeevfd) [reply helm-chart/values.yaml:327]: @azhavoro The problem with default paths with predefined host "cvat.local" that if you specify your own host and other configs for paths, then you get:
```yaml
hosts:
  - host: cvat.local
     paths:
       - path: /api
   - host: my-cvat-host.example.com
      paths:
        - path: /api
```
It is a bit ugly and contains unused values, I think it'd be better to preserve default paths maybe and allow to specify host (as basic config)

Comment (eremeevfd) [reply helm-chart/values.yaml:327]: Also, according to this docs: https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/basic-configuration/#new-features-available-in-kubernetes-118-and-above
ImplementationSpecific is a default value, so we can just omit it here

Comment (Keramblock) [reply helm-chart/values.yaml:330]: @eremeevfd default values is for cvat.local host. I doubt that will be usefull for other users.

Comment (Keramblock) [reply helm-chart/values.yaml:330]: > @azhavoro @Keramblock Maybe we should add ImplementationSpecific to default values?

default values is for cvat.local host. I doubt that will be usefull for other users.

File: helm-chart/values.yaml (hunk 4 above)
Comment (Keramblock) [reply helm-chart/values.yaml]: done

**Review by eremeevfd — APPROVED** [review]

**Comment by nmanovic** [comment]
@Keramblock , thanks for the contribution! Could you please fix remark linter? You use `[helm]` prefix in CHANGELOG.md. It makes the linter unhappy. You need to escape `[ ]` or remove the prefix.

**Comment by Keramblock** [comment]
> @Keramblock , thanks for the contribution! Could you please fix remark linter? You use `[helm]` prefix in CHANGELOG.md. It makes the linter unhappy. You need to escape `[ ]` or remove the prefix.

sure, will fix that in 10 minutes

**Comment by Keramblock** [comment]
Done

**Comment by Keramblock** [comment]
@nmanovic @azhavoro @mdacoca could you please take a look?

**Comment by Keramblock** [comment]
@nmanovic @azhavoro @mdacoca am I doing something wrong? I trying to help, but it looks like no one is interested in that.

**Comment by azhavoro** [comment]
> @nmanovic @azhavoro @mdacoca am I doing something wrong? I trying to help, but it looks like no one is interested in that.

@Keramblock Hi, sorry for the delay, this is great PR, but the helm tests are fading, I'm going to take a look, but I'm a little busy right now.

**Review by DreamingRaven — APPROVED** [review]
allgood
I wish I could help here at all, but for what its worth, everything seems to be working on my end!
I hope they get around to this soon!

**Comment by Keramblock** [comment]
Thanks a lot!

**Comment by Keramblock** [comment]
@azhavoro any news?

**Comment by nmanovic** [comment]
@Keramblock , could you please resolve the current conflicts? Please also look at problems with the linter.

**Comment by Keramblock** [comment]
@nmanovic  I tried to fix helm checks, but it sadly didnt succeed in that with following error:
` Command failed: kubectl exec release-5598438317-1-backend-server-8568c4654b-ztdkz -- tar --strip 3 -xjf /tmp/cvat_data.tar.bz2 -C /home/django/data/.

Error message: Defaulted container "cvat-app-backend-server-container" out of: cvat-app-backend-server-container, user-data-permission-fix (init)
tar: cache: Cannot utime: Operation not permitted
tar: cache: Cannot change mode to rwxr-xr-x: Operation not permitted`
I have limited knowledge of github ci and your CI infrastructure, so I need someones help with that.

**Comment by Keramblock** [comment]
I think I found source of problems - looks like that is old values file, that for some reason located outside of chart dir, that was not updated. I will try to fix it

**Comment by Keramblock** [comment]
That looks like last PR to develop just broke docker build

**Comment by kirill-sizov** [comment]
> That looks like last PR to develop just broke docker build

Hi, @Keramblock Actually that [last PR](https://github.com/opencv/cvat/pull/6514) isn't related to Helm tests at all. It seems that it was a temporary connection problem since as you can see in the [log](https://github.com/opencv/cvat/actions/runs/5599098812/jobs/10239618997) builder wasn't able to download the library from GitHub.

I've restarted the workflow for this PR let's see how it goes

**Comment by Keramblock** [comment]
Thanks a lot for the help!

**Comment by azhavoro** [comment]
@Keramblock Hi, there is a patch that fixes our tests
[fix.zip](https://github.com/opencv/cvat/files/12142849/fix.zip)

**Comment by Keramblock** [comment]
> @Keramblock Hi, there is a patch that fixes our tests [fix.zip](https://github.com/opencv/cvat/files/12142849/fix.zip)

@azhavoro thanks a lot! will commit that today.

**Comment by Keramblock** [comment]
@azhavoro  I think it is ready for review

**Review by azhavoro — APPROVED** [review]

**Comment by Keramblock** [comment]
@azhavoro @nmanovic @kirill-sizov  thanks a lot for the help!
